core/sequencer/server.go

```diff
@@ -222,72 +223,58 @@ func (s *Server) unappliedMetric(ctx context.Context, directoryID string, maxCou
 	return nil
 }
 
-// RunBatch runs the full sequence of steps (for one directory) nessesary to get a
-// mutation from the log integrated into the map. This consists of a series of
-// idempotent steps:
-// a) assign a batch of mutations from the logs to a map revision
-// b) apply the batch to the map
-// c) publish existing map roots to a log of SignedMapRoots.
+// RunBatch builds multiple outstanding revisions of a single directory's map
+// by integrating the corresponding mutations.
+//
+// TODO(pavelkalinnikov): Name it properly.
 func (s *Server) RunBatch(ctx context.Context, in *spb.RunBatchRequest) (*empty.Empty, error) {
-	defResp, err := s.loopback.DefineRevisions(ctx, &spb.DefineRevisionsRequest{
-		DirectoryId: in.DirectoryId,
-		MinBatch:    in.MinBatch,
-		MaxBatch:    in.MaxBatch})
+	revs, err := s.loopback.GetUnappliedRevisions(ctx,
+		&spb.GetUnappliedRevisionsRequest{DirectoryId: in.DirectoryId})
 	if err != nil {
 		return nil, err
 	}
 
-	unappliedRevisions.Set(float64(len(defResp.OutstandingRevisions)))
+	count, end := revs.End-revs.Begin, revs.End
```

Comment:
This changes the behavior slightly -- could this be it's own PR?

Reply:
I think the metric should be more or less the same.